sopht_mpi/utils/mpi_utils_2d.py

```diff
@@ -114,99 +115,115 @@ def __init__(self, ghost_size, mpi_construct):
         )
         self.column_type.Commit()
 
-        # non-blocking comm, stuff
-        self.num_requests = (
-            mpi_construct.grid_dim * 2 * 2
-        )  # dimension * 2 request for send/recv * 2 directions along each axis
-        # Initialize the requests array?
-        self.comm_requests = [
-            0,
-        ] * self.num_requests
+        # Initialize requests list for non-blocking comm
+        self.comm_requests = []
 
-    def exchange_init(self, local_field, mpi_construct):
+    def exchange_scalar_field_init(self, local_field):
         """
-        Exchange ghost data between neighbors.
+        Exchange scalar field ghost data between neighbors.
         """
         # Lines below to make code more literal
         y_axis = 0
         x_axis = 1
```

Comment:
replace with x_axis_idx and y_axis_idx from vectorfield

Reply:
This is something that I thought of a few weeks ago [here](https://github.com/fankiat/sopht-mpi/issues/92#issuecomment-1331427349), but decided not to do anything yet at that point. The `x_axis` here is not the vector field x-component, rather it's the grid topology x-axis.

Reply:
makes sense.